stardog/connection.py

```diff
@@ -77,7 +82,7 @@ def begin(self):
               If already in a transaction
         """
         self._assert_not_in_transaction()
-        self.transaction = self.conn.begin()
+        self.transaction = self.conn.begin(reasoning)
```

Comment:
Can we make this a non-positional argument?  `(reasoning=reasoning)`

Reply:
I added another commit that makes it a keyword arg. Now connection.begin() passes its keyword args (including reasoning) on as the params for the HTTP POST request to the server.